core/lib/merkle_tree/src/consistency.rs

```diff
@@ -69,10 +69,17 @@ pub enum ConsistencyError {
 impl<DB: Database, H: HashTree> MerkleTree<DB, H> {
     /// Verifies the internal tree consistency as stored in the database.
     ///
+    /// If `validate_indices` flag is set, it will be checked that indices for all tree leaves are unique
+    /// and are sequentially assigned starting from 1.
+    ///
```

Comment:
This check was there previously, it's just that it's optional now because we cannot guarantee that the index assignment will be valid locally in the Merkle tree logic.